### PR TITLE
Add option to use custom pruning module

### PR DESCRIPTION
**Pull request by MarshallOfSound**
* [x] I have read the [contribution documentation](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] The changes are appropriately documented (if applicable).
* [x] The changes have sufficient test coverage (if applicable).
* [x] The testsuite passes successfully on my local machine (if applicable).

**Summarize your changes:**

This should help avoid package manager specific pruning issues as we now walk the dependency tree manually.

### CODE REVIEW

File: prune.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const child = require('child_process')
 const debug = require('debug')('electron-packager')
+const { Walker } = require('pruner')
```
Comment (malept) [inline prune.js]: Packager is Node 4-compatible :smile:

File: docs/api.md [hunk 2]
```diff
@@ -228,7 +228,7 @@ Whether to replace an already existing output directory for a given platform (`t
 
 ##### `packageManager`
 
-*String* (default: `npm`)
+*String* | *Boolean* (default: `npm`)
```
Comment (malept) [inline docs/api.md]: In other places, we use `or`.

**Comment by malept** [comment]
There should probably be a test added too.

**Comment by malept** [comment]
@MarshallOfSound pruner itself is not Node 4 compatible: https://travis-ci.org/electron-userland/electron-packager/jobs/257800435#L2470

**Comment by MarshallOfSound** [comment]
@malept Changed the `tsc` target and it should be v4 compatible now 👍

**Comment by MarshallOfSound** [comment]
@malept The failure here look unrelated to the code changes, is it gtg?

**Pull request merged**